test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.ConnectionLifecycle.cs

```diff
@@ -23,11 +24,18 @@ public class ConnectionLifecycle
 
             public static IEnumerable<object[]> MethodsNamesThatRequireActiveConnection => MethodsThatRequireActiveConnection.Keys.Select(k => new object[] { k });
 
+            private HubConnection CreateHubConnection(Func<IConnection> connectionFactory)
+            {
+                IServiceCollection serviceCollection = new ServiceCollection();
+                var serviceProvider = serviceCollection.BuildServiceProvider();
+                return new HubConnection(connectionFactory, new JsonHubProtocol(), serviceProvider);
```

Comment:
This looks messed up. Now I think we need to go further here and stick the HubConnection itself in the container so we can just resolve it.

Reply:
It doesn't matter here because it is a test, but do you want HubConnectionBuilder to use the container to create the HubConnection? Reflection, etc, will be involved.